Process factory

kress95 opened the pull request:
#74 depends on this

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/hackerexperience/helix/77)
<!-- Reviewable:end -->

mememori commented:
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 18 unresolved discussions.

---

*[apps/helix_process/lib/process/model/process.ex, line 57 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRItodbeL0V8BxRbZR:-KgRItodbeL0V8BxRbZS:b3s30a9) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/lib/process/model/process.ex#L57)):*
> ```Elixir
> 
>     # FIXME: this was just added to allow producing the factory
>     field :software, :map
> ```

What ? This was an old field that i removed before (and it seems that, for some reason, the migration didn't go upstream). This became `process_data`

---

*[apps/helix_process/lib/process/model/process.ex, line 112 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJ3ko8MAXxzFWjujO:-KgRJ3ko8MAXxzFWjujP:b-pi9d9n) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/lib/process/model/process.ex#L112)):*
> ```Elixir
>     target_server_id
>     file_id
>     software
> ```

remove this

---

*[apps/helix_process/priv/repo/migrations/20170329222422_add_process_data.exs, line 6 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJ5JrTu1o9Kgi3Ns1:-KgRJ5JrTu1o9Kgi3Ns2:b-9ykdkr) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/priv/repo/migrations/20170329222422_add_process_data.exs#L6)):*
> ```Elixir
>   def change do
>     alter table(:processes) do
>       add :process_data, :jsonb
> ```

`null: false`

---

*[apps/helix_process/test/controller/process_test.exs, line 23 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJR9EJxmh9TAIlQVu:-KgRJEePdADjCMivWqTv:b-4ppegx) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L23)):*
> ```Elixir
>       software: %{}
>     }
>   end
> ```

As i said before, i find this extremely hard to read. You either want to use params (then you should make them explicit at the test) or use a record (then you should use the factory).

---

*[apps/helix_process/test/controller/process_test.exs, line 38 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJio23QWikbMc5L8K:-KgRJio23QWikbMc5L8L:beugv87) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L38)):*
> ```Elixir
>     end
> 
>     test "fails when account with id doesn't exist" do
> ```

What ????

---

*[apps/helix_process/test/controller/process_test.exs, line 43 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJnhrtnko8LuNSYW2:-KgRJnhrtnko8LuNSYW3:bijlvlu) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L43)):*
> ```Elixir
>   end
> 
>   describe "deleting" do
> ```

`describe "delete/1" do`

---

*[apps/helix_process/test/controller/process_test.exs, line 53 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJqOs9IcmZLaYd_Be:-KgRJqOs9IcmZLaYd_Bf:bve2q39) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L53)):*
> ```Elixir
>     end
> 
>     test "can be done by its id" do
> ```

`accepts id`

---

*[apps/helix_process/test/controller/process_test.exs, line 61 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJrLrxqooSjPRuaa1:-KgRJrLrxqooSjPRuaa2:b-bfquty) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L61)):*
> ```Elixir
>     end
> 
>     test "can be done by its struct" do
> ```

`accepts process struct`

---

*[apps/helix_process/test/support/factory.ex, line 28 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRKB3mng2Bzal0uGJ_:-KgRKB3mng2Bzal0uGJa:b-47ceo4) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L28)):*
> ```Elixir
> 
>     %Process{
>       gateway_id: PK.pk_for(Process),
> ```

This is incorrect and didn't i tell you not to use this ?

---

*[apps/helix_process/test/support/factory.ex, line 29 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRKAnEqPxc2prCbhvO:-KgRKAnEqPxc2prCbhvP:b-ucxx1o) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L29)):*
> ```Elixir
>     %Process{
>       gateway_id: PK.pk_for(Process),
>       target_server_id: PK.pk_for(Server),
> ```

Didn't i tell you not to use this ?

---

*[apps/helix_process/test/support/factory.ex, line 30 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRK9GN7iFF-sen5msG:-KgRK9GN7iFF-sen5msH:b-ucxx1o) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L30)):*
> ```Elixir
>       gateway_id: PK.pk_for(Process),
>       target_server_id: PK.pk_for(Server),
>       file_id: PK.pk_for(File),
> ```

Didn't i tell you not to use this ?

---

*[apps/helix_process/test/support/factory.ex, line 32 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRK88Ttwr_a1b-lK8C:-KgRK88Ttwr_a1b-lK8D:b-cm0t46) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L32)):*
> ```Elixir
>       file_id: PK.pk_for(File),
>       network_id: Random.pk(),
>       software: %{},
> ```

this should not exist

---

*[apps/helix_process/test/support/factory.ex, line 35 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRK4oMC1CAr6denmwt:-KgRK4oMC1CAr6denmwu:bb6ghje) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L35)):*
> ```Elixir
>       process_type: Random.string(min: 20, max: 20),
>       state: random_process_state(),
>       priority: Random.number(0..3),
> ```

0..5

---

*[apps/helix_process/test/support/factory.ex, line 36 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRK1QT8wlkh8pkbQBL:-KgRK1QT8wlkh8pkbQBM:bksh5v0) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L36)):*
> ```Elixir
>       state: random_process_state(),
>       priority: Random.number(0..3),
>       processed: %{},
> ```

This is incorrect

---

*[apps/helix_process/test/support/factory.ex, line 37 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRK1AUfwmfuIQ6rrfI:-KgRK1AUfwmfuIQ6rrfJ:bksh5v0) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L37)):*
> ```Elixir
>       priority: Random.number(0..3),
>       processed: %{},
>       allocated: %{},
> ```

This is incorrect

---

*[apps/helix_process/test/support/factory.ex, line 38 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRK-v21Oh7UKDYlAuE:-KgRK-v21Oh7UKDYlAuF:bksh5v0) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L38)):*
> ```Elixir
>       processed: %{},
>       allocated: %{},
>       limitations: %{},
> ```

This is incorrect

---

*[apps/helix_process/test/support/factory.ex, line 39 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJxx0ZuPBypU70Ios:-KgRJxx0ZuPBypU70Iot:bp6vprb) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L39)):*
> ```Elixir
>       allocated: %{},
>       limitations: %{},
>       creation_time: now,
> ```

Burette has time generators

---

*[apps/helix_process/test/support/factory.ex, line 40 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRKEU4syzFLA302azO:-KgRKEU4syzFLA302azP:b-prilrn) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L40)):*
> ```Elixir
>       limitations: %{},
>       creation_time: now,
>       updated_time: now
> ```

this is okay

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

kress95 commented:
*[apps/helix_process/lib/process/model/process.ex, line 112 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJ3ko8MAXxzFWjujO:-KgU7Jfs4n_svSuNLVyN:b-g4y2vl) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/lib/process/model/process.ex#L112)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

remove this
</blockquote></details>

I don't know how tests passed with this, really.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

kress95 commented:
*[apps/helix_process/test/controller/process_test.exs, line 38 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJio23QWikbMc5L8K:-KgU7eTZwRL4PPIxO5aN:bczn76b) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L38)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

What ????
</blockquote></details>

I was using account tests as a template, and was kinda sleepy :|

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

kress95 commented:
*[apps/helix_process/test/controller/process_test.exs, line 43 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJnhrtnko8LuNSYW2:-KgU8cE7qMba2RcGA2jV:b-5zz3y6) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L43)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

`describe "delete/1" do`
</blockquote></details>

so we're going back to that style?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

kress95 commented:
*[apps/helix_process/lib/process/model/process.ex, line 112 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJ3ko8MAXxzFWjujO:-KgUZJbxYPyNnQrLYLkH:bg8p341) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/lib/process/model/process.ex#L112)):*
<details><summary><i>Previously, kress95 (Christian Ferraz Lemos de Sousa) wrote…</i></summary><blockquote>

I don't know how tests passed with this, really.
</blockquote></details>

Wait, I was doidão and got this out of context, I didn't see that there was a `@creation_fields ~w/` above it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

kress95 commented:
Review status: 1 of 5 files reviewed at latest revision, 19 unresolved discussions.

---

*[apps/helix_process/priv/repo/migrations/20170329222422_add_process_data.exs, line 6 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJ5JrTu1o9Kgi3Ns1:-KgUJbFgK1E65VXX4A72:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/priv/repo/migrations/20170329222422_add_process_data.exs#L6)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

`null: false`
</blockquote></details>

Done.

---

*[apps/helix_process/test/controller/process_test.exs, line 23 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJR9EJxmh9TAIlQVu:-KgUYglJkchBR63luyLm:b1nyrbi) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L23)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

As i said before, i find this extremely hard to read. You either want to use params (then you should make them explicit at the test) or use a record (then you should use the factory).
</blockquote></details>

I don't know if it's done right, but  I changed it

---

*[apps/helix_process/test/controller/process_test.exs, line 38 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJio23QWikbMc5L8K:-KgUY0ioCR5fb3S2uJPa:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L38)):*
<details><summary><i>Previously, kress95 (Christian Ferraz Lemos de Sousa) wrote…</i></summary><blockquote>

I was using account tests as a template, and was kinda sleepy :|
</blockquote></details>

Done.

---

*[apps/helix_process/test/controller/process_test.exs, line 43 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJnhrtnko8LuNSYW2:-KgUXVEpjNv7LTmvkxyb:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L43)):*
<details><summary><i>Previously, kress95 (Christian Ferraz Lemos de Sousa) wrote…</i></summary><blockquote>

so we're going back to that style?
</blockquote></details>

Done.

---

*[apps/helix_process/test/controller/process_test.exs, line 53 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJqOs9IcmZLaYd_Be:-KgUXRmMLVvCDeyBlkZ3:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L53)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

`accepts id`
</blockquote></details>

Done.

---

*[apps/helix_process/test/controller/process_test.exs, line 61 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJrLrxqooSjPRuaa1:-KgUXRNbosYGAiAlqlN_:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L61)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

`accepts process struct`
</blockquote></details>

Done.

---

*[apps/helix_process/test/support/factory.ex, line 28 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRKB3mng2Bzal0uGJ_:-KgUJeIr0T2qDQxitUu4:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L28)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

This is incorrect and didn't i tell you not to use this ?
</blockquote></details>

Done.

---

*[apps/helix_process/test/support/factory.ex, line 29 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRKAnEqPxc2prCbhvO:-KgUJeo1YxC-ZzFVwcfd:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L29)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

Didn't i tell you not to use this ?
</blockquote></details>

Done.

---

*[apps/helix_process/test/support/factory.ex, line 30 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRK9GN7iFF-sen5msG:-KgUJfC8bSHofuEFBil8:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L30)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

Didn't i tell you not to use this ?
</blockquote></details>

Done.

---

*[apps/helix_process/test/support/factory.ex, line 32 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRK88Ttwr_a1b-lK8C:-KgUJg9pZemuv6u4BrAp:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L32)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

this should not exist
</blockquote></details>

Done.

---

*[apps/helix_process/test/support/factory.ex, line 35 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRK4oMC1CAr6denmwt:-KgUJhwF8VILFS1BMGuB:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L35)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

0..5
</blockquote></details>

Done.

---

*[apps/helix_process/test/support/factory.ex, line 36 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRK1QT8wlkh8pkbQBL:-KgUM-BqWbN_fx4DIOoQ:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L36)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

This is incorrect
</blockquote></details>

Done.

---

*[apps/helix_process/test/support/factory.ex, line 37 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRK1AUfwmfuIQ6rrfI:-KgULzL3KAP7PVij8vCf:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L37)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

This is incorrect
</blockquote></details>

Done.

---

*[apps/helix_process/test/support/factory.ex, line 38 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRK-v21Oh7UKDYlAuE:-KgULzkV4ZmiiFg0ehl3:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L38)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

This is incorrect
</blockquote></details>

Done.

---

*[apps/helix_process/test/support/factory.ex, line 39 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJxx0ZuPBypU70Ios:-KgUXW9XReV3qj-OkS2X:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L39)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

Burette has time generators
</blockquote></details>

Done.

---

*[apps/helix_process/test/support/factory.ex, line 40 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRKEU4syzFLA302azO:-KgUM0kUmkCkvX3uFFRI:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L40)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

this is okay
</blockquote></details>

Done.

---

*[apps/helix_process/lib/process/model/process.ex, line 57 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgUJM2c-w_NPG1hvi2O:-KgUJM2dohO_8E4VRye4:b5vo9jv) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/lib/process/model/process.ex#L57)):*
> null: false

---

*[apps/helix_process/lib/process/model/process.ex, line 57 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRItodbeL0V8BxRbZR:-KgUJZw0QWN9hssv-FoO:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/lib/process/model/process.ex#L57)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

What ? This was an old field that i removed before (and it seems that, for some reason, the migration didn't go upstream). This became `process_data`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

kress95 commented:
*[apps/helix_process/lib/process/model/process.ex, line 57 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgUJM2c-w_NPG1hvi2O:-KgUhRZjWrHiB8oV4_R0:b3jp6o0) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/lib/process/model/process.ex#L57)):*
<details><summary><i>Previously, kress95 (Christian Ferraz Lemos de Sousa) wrote…</i></summary><blockquote>

> null: false
</blockquote></details>

accidental mis-quote, reviewable doesn't allow removing it 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

mememori commented:
Reviewed 5 of 5 files at r2.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[apps/helix_process/test/controller/process_test.exs, line 23 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJR9EJxmh9TAIlQVu:-KgV4z8k6WWWNpjAZ0hm:ba8smzu) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L23)):*
<details><summary><i>Previously, kress95 (Christian Ferraz Lemos de Sousa) wrote…</i></summary><blockquote>

I don't know if it's done right, but  I changed it
</blockquote></details>

http://blog.plataformatec.com.br/2014/04/improve-your-test-readability-using-the-xunit-structure/

It's better but not ideal yet. I know the form the `params` should have but have no idea on what they are. And this is important because DRY is not relevant on tests, readability and context are.

And this will bite you when you use a pre-generated format but usually your entry point might produce different input. Eg: you use `Factory.params_for` as if it was the input on an external controller

---

*[apps/helix_process/test/controller/process_test.exs, line 38 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJio23QWikbMc5L8K:-KgV62Pq67h2zFjNgVEm:boieen6) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L38)):*
<details><summary><i>Previously, kress95 (Christian Ferraz Lemos de Sousa) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Still not good. An id is the very identity of a process, if it doesn't exists it means the process doesn't exist not that "process with id doesn't exist"

---

*[apps/helix_process/test/controller/process_test.exs, line 43 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJnhrtnko8LuNSYW2:-KgV6DaThy_8Cs2q6bmy:b-v5h81t) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L43)):*
<details><summary><i>Previously, kress95 (Christian Ferraz Lemos de Sousa) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

If you are testing a function, it's the most adequate. We just weren't enforcing it, so you followed a style and i did follow another

---

*[apps/helix_process/test/support/factory.ex, line 39 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJxx0ZuPBypU70Ios:-KgV6iO-4NSx00s8w823:b-jqhxzl) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L39)):*
<details><summary><i>Previously, kress95 (Christian Ferraz Lemos de Sousa) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Are you implying that a process can be created _in the future_ ? :joy:

---

*[apps/helix_process/test/support/factory.ex, line 35 at r2](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgV6oiUJb0Ia6ErGr1f:-KgV6oiUJb0Ia6ErGr1g:b-8vc4q2) ([raw file](https://github.com/hackerexperience/helix/blob/85266ac73c389b09c96ae2cf9220a3787e6e78c5/apps/helix_process/test/support/factory.ex#L35)):*
> ```Elixir
>       priority: Random.number(0..5),
>       objective: build(:resources),
>       processed: build(:resources),
> ```

This is okay but it would be better if you MAX-ed it to ensure that processed is never bigger than objective as this could cause bad state

---

*[apps/helix_process/test/support/factory.ex, line 36 at r2](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgV72VCEB7GRCes79Ag:-KgV72VCEB7GRCes79Ah:b-hhu1lz) ([raw file](https://github.com/hackerexperience/helix/blob/85266ac73c389b09c96ae2cf9220a3787e6e78c5/apps/helix_process/test/support/factory.ex#L36)):*
> ```Elixir
>       objective: build(:resources),
>       processed: build(:resources),
>       allocated: build(:resources),
> ```

This is okay but it would be better if you MAX-ed it to ensure that the amount allocated is never bigger than it's limit

---

*[apps/helix_process/test/support/factory.ex, line 48 at r2](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgV6z_DTbkMsPA_6SsT:-KgV6z_DTbkMsPA_6SsU:b-k0zy3k) ([raw file](https://github.com/hackerexperience/helix/blob/85266ac73c389b09c96ae2cf9220a3787e6e78c5/apps/helix_process/test/support/factory.ex#L48)):*
> ```Elixir
>       ram: Random.number(0..1024),
>       dlk: Random.number(0..100),
>       ulk: Random.number(0..100)
> ```

Too low range

---

*[apps/helix_process/test/support/factory.ex, line 58 at r2](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgV7wJz-TOp7V4z_vwA:-KgV7wJz-TOp7V4z_vwB:b-cdi0r4) ([raw file](https://github.com/hackerexperience/helix/blob/85266ac73c389b09c96ae2cf9220a3787e6e78c5/apps/helix_process/test/support/factory.ex#L58)):*
> ```Elixir
>       dlk: Random.number(100..1000),
>       ulk: Random.number(100..1000)
>     }
> ```

Wrong, a limited field can (and should) be nil most of the times (meaning it is not limited.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

kress95 commented:
*[apps/helix_process/test/support/factory.ex, line 39 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJxx0ZuPBypU70Ios:-KgVBIA7_6Dg__Xt7Lvx:b61anto) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/support/factory.ex#L39)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

Are you implying that a process can be created _in the future_ ? :joy:
</blockquote></details>

Damn time travelling computer processes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

kress95 commented:
*[apps/helix_process/test/support/factory.ex, line 48 at r2](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgV6z_DTbkMsPA_6SsT:-KgWSxxTgGT98S7gt0zM:b-n5msen) ([raw file](https://github.com/hackerexperience/helix/blob/85266ac73c389b09c96ae2cf9220a3787e6e78c5/apps/helix_process/test/support/factory.ex#L48)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

Too low range
</blockquote></details>

I'll make it be over nine thousand.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

kress95 commented:
Review status: 3 of 5 files reviewed at latest revision, 7 unresolved discussions.

---

*[apps/helix_process/test/controller/process_test.exs, line 23 at r1](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgRJR9EJxmh9TAIlQVu:-Kg_OVdEk2pqCr7i9t6X:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/06dd62d8659aca1a29983c28c359bd49f989d488/apps/helix_process/test/controller/process_test.exs#L23)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

http://blog.plataformatec.com.br/2014/04/improve-your-test-readability-using-the-xunit-structure/

It's better but not ideal yet. I know the form the `params` should have but have no idea on what they are. And this is important because DRY is not relevant on tests, readability and context are.

And this will bite you when you use a pre-generated format but usually your entry point might produce different input. Eg: you use `Factory.params_for` as if it was the input on an external controller
</blockquote></details>

Done.

---

*[apps/helix_process/test/support/factory.ex, line 48 at r2](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgV6z_DTbkMsPA_6SsT:-Kg_KbsqiE2Zz0-SFPlx:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/85266ac73c389b09c96ae2cf9220a3787e6e78c5/apps/helix_process/test/support/factory.ex#L48)):*
<details><summary><i>Previously, kress95 (Christian Ferraz Lemos de Sousa) wrote…</i></summary><blockquote>

I'll make it be over nine thousand.
</blockquote></details>

Done.

---

*[apps/helix_process/test/support/factory.ex, line 58 at r2](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgV7wJz-TOp7V4z_vwA:-Kg_KN56CJZyv5M1U5J0:b-896fix) ([raw file](https://github.com/hackerexperience/helix/blob/85266ac73c389b09c96ae2cf9220a3787e6e78c5/apps/helix_process/test/support/factory.ex#L58)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

Wrong, a limited field can (and should) be nil most of the times (meaning it is not limited.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

kress95 commented:
Reason for not updating descriptions:

No update were done for controller tests (at least until now), so I'm assuming this should be delayed.
Also, I'm not 100% sure about which format they should have, we don't have any documentation for that conventions.

mememori commented:
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[apps/helix_process/test/support/factory.ex, line 33 at r3](https://reviewable.io:443/reviews/hackerexperience/helix/77#-Kg_rKhmYHFTmoGQ05Ig:-Kg_rKhmYHFTmoGQ05Ih:b-c5gyo0) ([raw file](https://github.com/hackerexperience/helix/blob/0ed9b2946f89b60e952f736c08f319ace3bcf237/apps/helix_process/test/support/factory.ex#L33)):*
> ```Elixir
> 
>     build(:resources, params)
>   end
> ```

same problems from the other PR

---

*[apps/helix_process/test/support/factory.ex, line 44 at r3](https://reviewable.io:443/reviews/hackerexperience/helix/77#-Kg_rN6O9A_L5vYRUpx8:-Kg_rN6O9A_L5vYRUpx9:b-c5gyo0) ([raw file](https://github.com/hackerexperience/helix/blob/0ed9b2946f89b60e952f736c08f319ace3bcf237/apps/helix_process/test/support/factory.ex#L44)):*
> ```Elixir
> 
>     build(:resources, params)
>   end
> ```

same problems from the other PR

---

*[apps/helix_process/test/support/factory.ex, line 72 at r3](https://reviewable.io:443/reviews/hackerexperience/helix/77#-Kg_rSXGpypYJ3e4idcC:-Kg_rSXGpypYJ3e4idcD:b-c5gyo0) ([raw file](https://github.com/hackerexperience/helix/blob/0ed9b2946f89b60e952f736c08f319ace3bcf237/apps/helix_process/test/support/factory.ex#L72)):*
> ```Elixir
>       ulk: Random.number(4096..8192)
>     }
>   end
> ```

same problems from the other PR

---

*[apps/helix_process/test/support/factory.ex, line 81 at r3](https://reviewable.io:443/reviews/hackerexperience/helix/77#-Kg_rUdXQ-zZXdKTJrSw:-Kg_rUdXQ-zZXdKTJrSx:b-k3ibgc) ([raw file](https://github.com/hackerexperience/helix/blob/0ed9b2946f89b60e952f736c08f319ace3bcf237/apps/helix_process/test/support/factory.ex#L81)):*
> ```Elixir
>       ulk: nil
>     }
>   end
> ```

same problems from the other PR

Also, a limitation is not always nil but ok

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

mememori commented:
Please don't move a PR to the "Review/QA" pipeline until you resolve all issues

kress95 commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[apps/helix_process/test/support/factory.ex, line 35 at r2](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgV6oiUJb0Ia6ErGr1f:-Kgadmha5PSqWXuChSpW:b-r70pit) ([raw file](https://github.com/hackerexperience/helix/blob/85266ac73c389b09c96ae2cf9220a3787e6e78c5/apps/helix_process/test/support/factory.ex#L35)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

This is okay but it would be better if you MAX-ed it to ensure that processed is never bigger than objective as this could cause bad state
</blockquote></details>

Solved on another revision.

---

*[apps/helix_process/test/support/factory.ex, line 36 at r2](https://reviewable.io:443/reviews/hackerexperience/helix/77#-KgV72VCEB7GRCes79Ag:-KgadojAC2FrK1R30z3Q:b-r70pit) ([raw file](https://github.com/hackerexperience/helix/blob/85266ac73c389b09c96ae2cf9220a3787e6e78c5/apps/helix_process/test/support/factory.ex#L36)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

This is okay but it would be better if you MAX-ed it to ensure that the amount allocated is never bigger than it's limit
</blockquote></details>

Solved on another revision.

---

*[apps/helix_process/test/support/factory.ex, line 33 at r3](https://reviewable.io:443/reviews/hackerexperience/helix/77#-Kg_rKhmYHFTmoGQ05Ig:-KgadkSfgU_9n1py6ZgR:b-r70pit) ([raw file](https://github.com/hackerexperience/helix/blob/0ed9b2946f89b60e952f736c08f319ace3bcf237/apps/helix_process/test/support/factory.ex#L33)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

same problems from the other PR
</blockquote></details>

Solved on another revision.

---

*[apps/helix_process/test/support/factory.ex, line 44 at r3](https://reviewable.io:443/reviews/hackerexperience/helix/77#-Kg_rN6O9A_L5vYRUpx8:-KgadlRrLFY787fHOThW:b-r70pit) ([raw file](https://github.com/hackerexperience/helix/blob/0ed9b2946f89b60e952f736c08f319ace3bcf237/apps/helix_process/test/support/factory.ex#L44)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

same problems from the other PR
</blockquote></details>

Solved on another revision.

---

*[apps/helix_process/test/support/factory.ex, line 72 at r3](https://reviewable.io:443/reviews/hackerexperience/helix/77#-Kg_rSXGpypYJ3e4idcC:-KgadsuUOedsor2Fwnyi:b-r70pit) ([raw file](https://github.com/hackerexperience/helix/blob/0ed9b2946f89b60e952f736c08f319ace3bcf237/apps/helix_process/test/support/factory.ex#L72)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

same problems from the other PR
</blockquote></details>

Solved on another revision.

---

*[apps/helix_process/test/support/factory.ex, line 81 at r3](https://reviewable.io:443/reviews/hackerexperience/helix/77#-Kg_rUdXQ-zZXdKTJrSw:-KgadtdBmEWaJUtQxbFk:bhiktaj) ([raw file](https://github.com/hackerexperience/helix/blob/0ed9b2946f89b60e952f736c08f319ace3bcf237/apps/helix_process/test/support/factory.ex#L81)):*
<details><summary><i>Previously, mememori (Charlotte '); DROP TABLE users;--) wrote…</i></summary><blockquote>

same problems from the other PR

Also, a limitation is not always nil but ok
</blockquote></details>

Solved on another revision, I'll leave it as nil for now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77)*
<!-- Sent from Reviewable.io -->

kress95 commented:
Okay, sorry, this is being continued on #74 

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/77#-:-Kgae-yMYwbA0Fwb6m4o:bbpfllx)*
<!-- Sent from Reviewable.io -->

kress95 commented:
Continuing on #74
